src/push/Permission.js

```diff
@@ -41,13 +41,17 @@ export default class Permission {
     _requestWithCallback(onGranted: () => void, onDenied: () => void) {
         const existing = this.get();
 
+        var resolved = false;
         var resolve = (result = this._win.Notification.permission) => {
+            if (resolved) return;
+            resolved = true;
             if (typeof result === 'undefined' && this._win.webkitNotifications)
                 result = this._win.webkitNotifications.checkPermission();
             if (result === this.GRANTED || result === 0) {
                 if (onGranted) onGranted();
             } else if (onDenied) onDenied();
         };
+        var request;
```

Comment:
I would moved this declaration to where `request` is actually defined. I believe the convention is you only define uninitialized variables in the top-most code block, and because this is used in a conditional block, its declaration should also be in that block.